skore/src/skore/sklearn/train_test_split.py

```diff
@@ -0,0 +1,252 @@
+"""Implement train_test_split."""
+
+from __future__ import annotations
+
+import warnings
+from typing import TYPE_CHECKING, Any, Optional, Union
+
+import pandas
+from numpy.random import RandomState
+
+from skore.project import Project
+
+if TYPE_CHECKING:
+    from skore.sklearn.cross_validate import MLTask
+
+    ArrayLike = Any
+
+
+def _find_ml_task(y) -> MLTask:
+    """Guess the ML task being addressed based on a target array.
+
+    Parameters
+    ----------
+    y : numpy.ndarray
+        A 1-dimensional target vector.
+
+    Returns
+    -------
+    Literal["binary-classification", "multiclass-classification",
+    "regression", "clustering", "unknown"]
+        The guess of the kind of ML task being performed.
+    """
+    import sklearn.utils.multiclass
+
+    if y is None:
+        # NOTE: The task might not be clustering
+        return "clustering"
+
+    type_of_target = sklearn.utils.multiclass.type_of_target(y)
+
+    if type_of_target == "binary":
+        return "binary-classification"
+
+    if type_of_target == "multiclass":
+        return "multiclass-classification"
+
+    if "continuous" in type_of_target:
+        return "regression"
+
+    if type_of_target == "unknown":
+        return "unknown"
+
+    return "unknown"
```

Comment:
Do you think you can factorize with `cross_validation._find_ml_task` ? 

```python
# sklearn/__init__.py

def guess_ml_task(y, estimator=None) -> MLTask:
```

Reply:
We discuss that and said that this may not be considered as a repetition of code until we need it in a third place ^^

Reply:
Putting it in `sklearn/__init__.py` would result in a circular import; where else could I put it? `sklearn/utils.py`?

Reply:
done